src/tree.rs

```diff
@@ -436,6 +438,63 @@ impl<T> Tree<T> {
     }
 
     ///
+    /// Swaps two `Node`s including their children given their `NodeId`s.
+    ///
+    /// Returns an empty `Result` containing a `NodeIdError` if one occurred on either provided `Id`.
+    ///
+    /// Notes:
+    ///     - Both `NodeId`s are still valid after this process and are not swapped.
+    ///     - Keeps the positions of the `Node`s in their parents children collection.
+    ///
+    /// ``
```

Comment:
I'm with you 100% on the `swap(B,C)` front.

As for the behavior described in `swap(C,F)` this would make more sense to me:
```
         A
      /     \
    B         C
   / \       / \
  D   E     F   G
 / \ / \   / \ / \
H  I J  K L  M N  O
```
would become:
```
         A
      /     \
    B         F
   / \       /|\
  D   E     C L M
 / \ / \     \
H  I J  K     G
             / \
            N   O
```
I'm thinking it should work that way because `F` was the first child of `C`, so `C` should now be the first child of `F`.  I also don't think order should matter here (calling `swap(C,F)` should be the same as calling `swap(F,C)`), but maybe I'm not thinking about that properly.  In addition, my first reaction is that `C`'s children shouldn't be dropped just because of a `swap`.

What do you think about that behavior?